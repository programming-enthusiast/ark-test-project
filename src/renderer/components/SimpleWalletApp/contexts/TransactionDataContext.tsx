import React from "react";

import { TransactionDataContextType } from "../types";

export default React.createContext<TransactionDataContextType>({
	isLoading: false,
	setLoading: () => null,
	wallets: [],
	setWallets: () => null,
	currentWalletIndex: 0,
	setCurrentWalletIndex: () => null,
});
