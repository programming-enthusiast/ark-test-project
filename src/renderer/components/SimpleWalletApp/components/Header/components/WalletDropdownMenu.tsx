import React, { useContext, useEffect, useState } from "react";

import WalletIcon from "../../../assets/Icons/address.svg";
import WalletIconHover from "../../../assets/Icons/address-hover.svg";
import TransactionDataContext from "../../../contexts/TransactionDataContext";

interface WalletDropdownMenuPropType {
	className: string;
}

const WalletDropdownMenu = ({ className }: WalletDropdownMenuPropType) => {
	const [isMenuOpen, setMenuOpen] = useState(false);
	const { wallets, currentWalletIndex, setCurrentWalletIndex } = useContext(
		TransactionDataContext
	);
	const [isOver, setIsOver] = useState(false);

	useEffect(() => {
		const handleClickOutside = (event: any) => {
			const dropdownMenu = document.getElementById(
				"wallet-dropdown-menu"
			);
			if (dropdownMenu && !dropdownMenu.contains(event.target)) {
				setMenuOpen(false);
			}
		};

		const handleMouseOver = (event: any) => {
			const dropdownMenu = document.getElementById(
				"dropdown-menu-wrapper"
			);
			if (dropdownMenu && dropdownMenu.contains(event.target)) {
				setIsOver(true);
			} else {
				setIsOver(false);
			}
		};

		// Bind the event listener
		document.addEventListener("mouseup", handleClickOutside);
		document.addEventListener("mouseover", handleMouseOver);
		return () => {
			// Unbind the event listener on clean up
			document.removeEventListener("mouseup", handleClickOutside);
			document.removeEventListener("mouseover", handleMouseOver);
		};
	}, []);

	const clickDropdownMenu = () => {
		console.log("clicked");
		setMenuOpen(!isMenuOpen);
	};

	const clickMenuItem = (index: number) => {
		console.log(index);
		setMenuOpen(false);
		setCurrentWalletIndex(index);
	};

	return (
		<div className={`${className} flex`} id="wallet-dropdown-menu">
			<div
				className="hidden sm:flex w-51px h-57px rounded-l-full"
				id="wallet-icon"
				style={{ background: isOver ? "#029383" : "#2F3333" }}
			>
				<img
					className={`${isOver ? "hidden" : ""} m-auto w-1/3 h-1/3`}
					src={WalletIcon}
				/>
				<img
					className={`${isOver ? "" : "hidden"} m-auto w-1/3 h-1/3`}
					src={WalletIconHover}
				/>
			</div>

			<div
				className="custom-select-wrapper w-247px md:w-461px lg:w-557px xl:w-302px xxl:w-404px"
				id="dropdown-menu-wrapper"
			>
				<div
					className={`custom-select w-full lg:w-557px xl:w-302px xxl:w-404px ${
						isMenuOpen ? "open" : ""
					}`}
				>
					<div
						className="custom-select__trigger"
						onClick={clickDropdownMenu}
					>
						<span className="overflow-hidden overflow-ellipsis">
							{wallets.length > 0 &&
								wallets[currentWalletIndex].address}
						</span>
						<div className="arrow" />
					</div>
					<div className="custom-options">
						{wallets.length > 0 &&
							wallets.map((wallet, index) => (
								<span
									className={
										"custom-option overflow-hidden overflow-ellipsis" +
										(index === currentWalletIndex
											? " selected"
											: "")
									}
									data-value={wallet.address}
									key={wallet.address}
									onClick={() => clickMenuItem(index)}
								>
									{wallet.address}
								</span>
							))}
					</div>
				</div>
			</div>
		</div>
	);
};

export default WalletDropdownMenu;
