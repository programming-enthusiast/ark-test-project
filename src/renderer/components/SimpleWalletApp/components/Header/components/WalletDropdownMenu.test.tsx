import React from "react";
import renderer from "react-test-renderer";

import TransactionDataContext from "../../../contexts/TransactionDataContext";
import WalletDropdownMenu from "./WalletDropdownMenu";

describe("WalletDropdownMenu Component", () => {
	let renderedComponent: any;

	const contextValue = {
		isLoading: false,
		currentWalletIndex: 0,
		wallets: [
			{
				address: "DR7WwjwGdw6mXoBrispUcTUHVV31AkAfs5",
				balance: 123456,
				transactions: [
					{
						txId: "efwfefweg",
						sender: "DTiBkzrfMjgS5PKKeZcorBvQ4PvV6me4F5",
						recipient: "DRHPX3fxEvXrKPmuy9RstC2gqNWdz5DDNk",
						timestamp: "22.10.2019",
						amount: 1234567,
						fee: 1234567,
					},
					{
						txId: "efwfefweg",
						sender: "DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES",
						recipient: "DRHPX3fxEvXrKPmuy9RstC2gqNWdz5DDNk",
						timestamp: "22.10.2019",
						amount: 1234567,
						fee: 1234567,
					},
					{
						txId: "efwfefweg",
						sender: "D6Z26L69gdk9qYmTv5uzk3uGepigtHY4ax",
						recipient: "DRHPX3fxEvXrKPmuy9RstC2gqNWdz5DDNk",
						timestamp: "22.10.2019",
						amount: 1234567,
						fee: 1234567,
					},
				],
			},
			{
				address: "D6Z26L69gdk9qYmTv5uzk3uGepigtHY4ax",
				balance: 123456,
				transactions: [
					{
						txId: "efwfefweg",
						sender: "DTiBkzrfMjgS5PKKeZcorBvQ4PvV6me4F5",
						recipient: "D6Z26L69gdk9qYmTv5uzk3uGepigtHY4ax",
						timestamp: "22.10.2019",
						amount: 1234567,
						fee: 1234567,
					},
				],
			},
			{
				address: "DNPBUxxGQUKDPX3XKUXa3pc4GK8yz7L97T",
				balance: 123456,
				transactions: [
					{
						txId: "efwfefweg",
						sender: "DTiBkzrfMjgS5PKKeZcorBvQ4PvV6me4F5",
						recipient: "DNPBUxxGQUKDPX3XKUXa3pc4GK8yz7L97T",
						timestamp: "22.10.2019",
						amount: 1234567,
						fee: 1234567,
					},
				],
			},
			{
				address: "DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES",
				balance: 123456,
				transactions: [
					{
						txId: "efwfefweg",
						sender: "DTiBkzrfMjgS5PKKeZcorBvQ4PvV6me4F5",
						recipient: "DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES",
						timestamp: "22.10.2019",
						amount: 1234567,
						fee: 1234567,
					},
				],
			},
			{
				address: "DRHPX3fxEvXrKPmuy9RstC2gqNWdz5DDNk",
				balance: 123456,
				transactions: [
					{
						txId: "efwfefweg",
						sender: "DTiBkzrfMjgS5PKKeZcorBvQ4PvV6me4F5",
						recipient: "DRHPX3fxEvXrKPmuy9RstC2gqNWdz5DDNk",
						timestamp: "22.10.2019",
						amount: 1234567,
						fee: 1234567,
					},
				],
			},
		],
	};

	beforeAll(() => {
		renderedComponent = renderer.create(
			<TransactionDataContext.Provider value={contextValue}>
				<WalletDropdownMenu className="mx-40px my-auto lg:m-auto" />
			</TransactionDataContext.Provider>
		);
	});

	it("should render correctly", () => {
		expect(renderedComponent.toJSON()).toMatchSnapshot();
	});
});
