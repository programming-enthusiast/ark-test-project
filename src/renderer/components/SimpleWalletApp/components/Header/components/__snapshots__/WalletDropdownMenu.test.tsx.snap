// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletDropdownMenu Component should render correctly 1`] = `
<div
  className="mx-40px my-auto lg:m-auto flex"
  id="wallet-dropdown-menu"
>
  <div
    className="hidden sm:flex w-51px h-57px rounded-l-full"
    id="wallet-icon"
    style={
      Object {
        "background": "#2F3333",
      }
    }
  >
    <img
      className=" m-auto w-1/3 h-1/3"
      src="address.svg"
    />
    <img
      className="hidden m-auto w-1/3 h-1/3"
      src="address-hover.svg"
    />
  </div>
  <div
    className="custom-select-wrapper w-247px md:w-461px lg:w-557px xl:w-302px xxl:w-404px"
    id="dropdown-menu-wrapper"
  >
    <div
      className="custom-select w-full lg:w-557px xl:w-302px xxl:w-404px "
    >
      <div
        className="custom-select__trigger"
        onClick={[Function]}
      >
        <span
          className="overflow-hidden overflow-ellipsis"
        >
          DR7WwjwGdw6mXoBrispUcTUHVV31AkAfs5
        </span>
        <div
          className="arrow"
        />
      </div>
      <div
        className="custom-options"
      >
        <span
          className="custom-option overflow-hidden overflow-ellipsis selected"
          data-value="DR7WwjwGdw6mXoBrispUcTUHVV31AkAfs5"
          onClick={[Function]}
        >
          DR7WwjwGdw6mXoBrispUcTUHVV31AkAfs5
        </span>
        <span
          className="custom-option overflow-hidden overflow-ellipsis"
          data-value="D6Z26L69gdk9qYmTv5uzk3uGepigtHY4ax"
          onClick={[Function]}
        >
          D6Z26L69gdk9qYmTv5uzk3uGepigtHY4ax
        </span>
        <span
          className="custom-option overflow-hidden overflow-ellipsis"
          data-value="DNPBUxxGQUKDPX3XKUXa3pc4GK8yz7L97T"
          onClick={[Function]}
        >
          DNPBUxxGQUKDPX3XKUXa3pc4GK8yz7L97T
        </span>
        <span
          className="custom-option overflow-hidden overflow-ellipsis"
          data-value="DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES"
          onClick={[Function]}
        >
          DRgF3PvzeGWndQjET7dZsSmnrc6uAy23ES
        </span>
        <span
          className="custom-option overflow-hidden overflow-ellipsis"
          data-value="DRHPX3fxEvXrKPmuy9RstC2gqNWdz5DDNk"
          onClick={[Function]}
        >
          DRHPX3fxEvXrKPmuy9RstC2gqNWdz5DDNk
        </span>
      </div>
    </div>
  </div>
</div>
`;
